### PR TITLE
[pkg/stanza] add splitter factory which return a split func

### PR DESCRIPTION
**Pull request by atingchen**
**Description:** <Describe what has changed.>

A new factory method has been added, and replace `helper.SplitterConfig` with `splitterFactory`.
**Link to tracking Issue:** <Issue number if applicable>
#14766
**Testing:** <Describe what testing was performed and which tests were added.>
Unit test

### CODE REVIEW

File: pkg/stanza/fileconsumer/splitter_factory.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		return nil, err
+	}
+	flusher := factory.Flusher.Build()
+	splitter, err := factory.Multiline.Build(enc.Encoding, false, flusher, maxLogSize)
+	if err != nil {
+		return nil, err
+	}
+	if flusher != nil {
+		splitter = flusher.SplitFunc(splitter)
+	}
```
Comment (djaglowski) [inline pkg/stanza/fileconsumer/splitter_factory.go:60]: I believe this is already done inside of `Multiline.Build`

Comment (atingchen) [reply pkg/stanza/fileconsumer/splitter_factory.go:60]: No, `Multiline.Build` just returns `SplitFunc`. This part of the work was previously implemented by `SplitterConfig.Build`.
Here we want to replace `SplitterConfig`.

Comment (djaglowski) [reply pkg/stanza/fileconsumer/splitter_factory.go:60]: Then why pass `flusher` into `factory.Multiline.Build`? It is used [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ad1c3778c43b40591b5c7a2c235f5ccccb958c04/pkg/stanza/operator/helper/multiline.go#L90-L92) to wrap the split func.

Comment (atingchen) [reply pkg/stanza/fileconsumer/splitter_factory.go:60]: Sorry, this is a duplicate code and I should remove it. I ignored it earlier.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**